### PR TITLE
CMake: Replace hard-coded install dirs with GNUInstallDirs

### PR DESCRIPTION
**Pull request by uklotzde**
https://bugzilla.rpmfusion.org/show_bug.cgi?id=5830

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1016,9 +1016,9 @@ elseif(UNIX)
   else()
     target_compile_definitions(mixxx-lib PRIVATE __UNIX__)
     target_compile_definitions(
-      mixxx-lib PUBLIC UNIX_SHARE_PATH="${CMAKE_INSTALL_PREFIX}/share/mixxx")
+      mixxx-lib PUBLIC UNIX_SHARE_PATH="${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_DATAROOTDIR}/mixxx")
     target_compile_definitions(
-      mixxx-lib PUBLIC UNIX_LIB_PATH="${CMAKE_INSTALL_PREFIX}/lib/mixxx")
+      mixxx-lib PUBLIC UNIX_LIB_PATH="${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/mixxx")
```
Comment (Holzhaus) [inline CMakeLists.txt:1021]: Is this even used? I grepped through the source but didn't find it.

File: CMakeLists.txt [hunk 2]
```diff
@@ -1016,9 +1016,9 @@ elseif(UNIX)
   else()
     target_compile_definitions(mixxx-lib PRIVATE __UNIX__)
     target_compile_definitions(
-      mixxx-lib PUBLIC UNIX_SHARE_PATH="${CMAKE_INSTALL_PREFIX}/share/mixxx")
+      mixxx-lib PUBLIC UNIX_SHARE_PATH="${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_DATAROOTDIR}/mixxx")
```
Comment (Holzhaus) [inline CMakeLists.txt:1019]: Can reorder code so that we can use the `MIXXX_INSTALL_DATADIR` variable from line 1045?

File: CMakeLists.txt (hunk 1 above)
Comment (uklotzde) [reply CMakeLists.txt:1021]: Then let's remove it, both from the SCons and CMake builds. I haven't checked how this is actually supposed to resolve the reported crash.

**Comment by Holzhaus** [comment]
Does it resolve the Fedora crash? If so, I think we can merge.

**Comment by uklotzde** [comment]
> Does it resolve the Fedora crash? If so, I think we can merge.

https://bugzilla.rpmfusion.org/show_bug.cgi?id=5830#c6

I don't think that this could have been the cause for the crash. Nevertheless it should not prevent us from cleaning up our build files ;)

**Review by Holzhaus — APPROVED** [review]
LGTM, thank you!

**Pull request merged**